### PR TITLE
Notify CCO on m.done, reorganise m.done / m.complete to use new shared mission completion functions

### LINKED ISSUE #248: Notify owner on m.done

### PR DESCRIPTION
**Pull request by cimspin**
Fixes #248 #241

### CODE REVIEW

File: MissionAlertBot.py [hunk 1]
```diff
@@ -1695,8 +1737,7 @@ async def remove_carrier_channel(mission_channel_id, seconds):
             await asyncio.wait_for(lock_mission_channel(), timeout=120)
         except asyncio.TimeoutError:
             print(f"No channel lock available for {delchannel}")
-            channel = bot.get_channel(bot_spam_id)
-            return await channel.send(f"<@211891698551226368> WARNING: No channel lock available on {delchannel} after 120 seconds. Deletion aborted.")
+            return await spamchannel.send(f"<@211891698551226368> WARNING: No channel lock available on {delchannel} after 120 seconds. Deletion aborted.")
```
Comment (gcruicks) [inline MissionAlertBot.py:1740]: hard coded user id here

File: MissionAlertBot.py [hunk 2]
```diff
@@ -1632,47 +1660,60 @@ async def done(ctx, carrier_name_search_term, rp=None):
...
+        # notify owner if not command author
+        carrier_data = find_carrier_from_long_name(mission_data.carrier_name)
+        if not ctx.author.id == carrier_data.ownerid:
+            # notify in channel - not sure this is needed anymore, leaving out for now
+            # await ctx.send(f"Notifying carrier owner: <@{carrier_data.ownerid}>")
+
+            # notify by DM
+            owner = await bot.fetch_user(carrier_data.ownerid)
+            if m_done:
+                if desc_msg: reason = f"  Reason given:\n{desc_msg}"
```
Comment (gcruicks) [inline MissionAlertBot.py:1713]: put this on a new line please :)

Comment (gcruicks) [reply MissionAlertBot.py:1713]: Also needs defined (reason) for the negative case

File: MissionAlertBot.py (hunk 1 above)
Comment (cimspin) [reply MissionAlertBot.py:1740]: That's me!

File: MissionAlertBot.py [hunk 3]
```diff
@@ -1632,47 +1660,60 @@ async def done(ctx, carrier_name_search_term, rp=None):
...
+                await submission.reply(reddit_complete_text)
                 # mark original post as spoiler, change its flair
                 await submission.flair.select(flair_mission_stop)
                 await submission.mod.spoiler()
             except:
                 await ctx.send("Failed updating Reddit :(")
 
         # delete mission entry from db
-        mission_db.execute(f'''DELETE FROM missions WHERE carrier LIKE (?)''', ('%' + carrier_name_search_term + '%',))
+        mission_db.execute(f"DELETE FROM missions WHERE carrier = '{mission_data.carrier_name}' ;")
```
Comment (gcruicks) [inline MissionAlertBot.py:1691]: do you need the semi-colon here?

File: MissionAlertBot.py [hunk 4]
```diff
@@ -1632,47 +1660,60 @@ async def done(ctx, carrier_name_search_term, rp=None):
...
+        # notify owner if not command author
+        carrier_data = find_carrier_from_long_name(mission_data.carrier_name)
+        if not ctx.author.id == carrier_data.ownerid:
+            # notify in channel - not sure this is needed anymore, leaving out for now
+            # await ctx.send(f"Notifying carrier owner: <@{carrier_data.ownerid}>")
+
+            # notify by DM
+            owner = await bot.fetch_user(carrier_data.ownerid)
+            if m_done:
+                reason = None if desc_msg == "" else desc_msg
```
Comment (gcruicks) [inline MissionAlertBot.py:1713]: `reason = desc_msg if desc_msg else None`, the positive case first makes it a bit more readable.

Also `if desc_msg == ''` can be caught with `if not desc_msg`

**Review by gcruicks — CHANGES_REQUESTED** [review]
minor comments only

File: MissionAlertBot.py (hunk 3 above)
Comment (cimspin) [reply MissionAlertBot.py]: Apparently not

File: MissionAlertBot.py (hunk 4 above)
Comment (cimspin) [reply MissionAlertBot.py]: Changes made

**Pull request merged**